pwa-shell/src/auth/default/DefaultAuthProvider.js

```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import DefaultAuthContext from './DefaultAuthContext';
+
+class DefaultAuthProvider extends Component {
+  render() {
+    const authInitialized = true; //there is no external auth to load
+    const auth = {
+      login: () =>
+        this.props.history.replace(
```

Comment:
why not using `history.push`?

Reply:
this is because if we use the push method, and the user chooses to go back (using back nav), it will always end up going back to login since the state before the login is, for example, protected content detail that is the user supposed to view but redirects to login for authentication. So my solution is, if unauthorized, it should replace the protected content detail to a login page so when they press back, it goes back to listing where it supposed to be